src/services/worker.service.ts

```diff
@@ -32,13 +33,13 @@ export class WorkerService {
    *
    * The resolved `Runner` object has a number of helpers on it, see [Runner object](https://github.com/graphile/worker#runner-object) for more information.
    */
-  async run(): Promise<void> {
+  async run(): Promise<Runner> {
     await this.runMigrations();
 
     this.logger.debug('Start runner');
 
     const runner = await run(this.options);
-    return runner.promise;
+    return runner;
```

Comment:
It's not exactly the same thing. We need to `await runner.promise` here
```suggestion
    await runner.promise;
    return runner;
```

Reply:
hm, to be honest, I don't think that is a correct fix; previously, you returned `runner.promise`; we didn't `await` it. And now, if i want to receive a runner, I need to write something like

```js
const runner = await app.get(WorkerService).run();
console.log(runner);
```
but because of the internal `await runner.promise` we will not see our `console.log`

from docs
```
promise: Promise<void> — a promise that resolves once the runner has completed.
```

Reply:
Ha sorry, i'll fix it and publish a new version

Reply:
cool, thank you

Reply:
I published a new version with this fix: https://github.com/madeindjs/nestjs-graphile-worker/commit/837d2d68a5acc493a32469f11be8aaf8d95c07db

Reply:
thank you